app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,58 @@
+class TweetsController < ApplicationController
+
+  get '/tweets' do
+    if logged_in?
+      erb :'/tweets/tweets'
+    else redirect '/login'
+    end
+  end
+
+  get '/tweets/new' do
+    if logged_in?
+      erb :'/tweets/create_tweet'
+    else
+      redirect '/login'
+    end
+  end
+
+  post '/tweets' do
+    redirect '/tweets/new' if params[:content].empty?
+    @tweet = Tweet.new(content: params[:content], user_id: current_user.id)
+    @tweet.save
+    erb :'/tweets/show_tweet'
+  end
+
+  get '/tweets/:id' do
+    redirect '/login' unless logged_in?
+    @tweet = Tweet.find_by_id(params[:id])
+    if @tweet
+      erb :'/tweets/show_tweet'
+    else
+      redirect '/tweets'
+    end
+  end
+
+  get '/tweets/:id/edit' do
+    redirect '/login' unless logged_in?
+    @tweet = Tweet.find_by_id(params[:id])
+    if @tweet
+      erb :'/tweets/edit_tweet'
+    else
+      redirect '/tweets'
+    end
+  end
+
+  patch '/tweets/:id/edit' do
+    @tweet = Tweet.find_by_id(params[:id])
+    @tweet.content = params[:content] unless params[:content].empty?
+    @tweet.save
+    erb :'/tweets/show_tweet'
+  end
+
+  delete '/tweets/:id/delete' do
+    @tweet = Tweet.find_by_id(params[:id])
+    @tweet.delete if @tweet.user_id == current_user.id
```

Comment:
Great job checking to make sure the tween belongs to the current user here! 

One thing to consider here is finding tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(id: params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully, simply returning nil if the user is trying to delete a tweet that they do not have access to.